src/js/select2/selection/base.js

```diff
@@ -65,12 +68,24 @@ define([
 
     container.on('selection:update', function (params) {
       self.update(params.data);
-    });
+      self.$selection.attr('aria-label', params.data.resultsId);
+      var $label = self.$selection.attr('aria-label');
+      var $labeltext;
+      var $rendered = self.$selection.find('.select2-selection__rendered');
+      var $title = $rendered.attr('title');
+      if ($title == undefined) {
+        $labeltext = $label + 'No value currently selected.';
+      }
+      else {
+      $labeltext = $label + 'The selected value is:' + $title;
+      }
+      self.$selection.attr('aria-label', $labeltext);
+      });
 
     container.on('open', function () {
       // When the dropdown is open, aria-expanded="true"
       self.$selection.attr('aria-expanded', 'true');
-      self.$selection.attr('aria-owns', resultsId);
+      self.$selection.attr('aria-owns', '.select2-container');
```

Comment:
Is this necessary, is it even allowed to provide a selector  rather than an id there?